src/main/java/org/jenkins/plugins/lockableresources/LockableResourcesManager.java

```diff
@@ -300,11 +377,33 @@ private synchronized void freeResources(List<String> unlockResourceNames, @Nulla
 						// No more contexts, unlock resource
 						resource.unqueue();
 						resource.setBuild(null);
+						// If the resource is on the stratos server delete the reservation objective
+						if (fromName(resource.getName(), stratosResources) !=null){
+							LOGGER.log(Level.FINE, "Freeing " + resource.getName() + " from " + getStratosURL());
```

Comment:
Same comment as above. Code that deals with the objective part of the LockableResource belongs in the LockableResource objective subclass, not in LockableResourceManager